notify-debouncer-mini/src/lib.rs

```diff
@@ -253,87 +289,91 @@ impl<T: Watcher> Debouncer<T> {
 impl<T: Watcher> Drop for Debouncer<T> {
     fn drop(&mut self) {
         // don't imitate c++ async futures and block on drop
-        self.set_stop();
+        self.send_stop();
```

Comment:
Dropping won't block and thus wait for the thread to finish. Stop does.

Reply:
But that's the point to clean up correctly, since you want to join on the thread? Otherwise you might exit before you drop the thread.

Reply:
Hm, I'm on the fence about this. The previous event loop would definitely hang for at most `timeout/4`, so this was just to prevent blocking on drop for no good reason.

The new passive debouncer shouldn't really block much, so we could join anyway. But with that the question arises: Do we even want to collect that thread ? We don't really need to, and notify has to spawn various threads for inotify anyway..

Reply:
But you have option to drop without any blocking at all with `stop_nonblocking`? I'd assume that watcher will also join and properly reap the inotify? Otherwise you'll leak the inotify nodes, which is not really good, since you might have an on-demand debouncer spawninig, or do some reconfigurations.

Doing proper resource management is what I'd expect from the `drop`(though, if you don't shutdown right away, you'll likely handle that just about fine).

Reply:
We won't leak anything, we can just decide whether or not to wait for the event loop to finish. We don't event need to create a join handle, we can just start the thread and let it end on its own, it should definitely end once a shutdown command is send, otherwise it'll not end anyway, we our drop will just lock up.